packages/server/src/server.ts

```diff
@@ -76,31 +76,119 @@ registerAdditionalApi(app, DbConnection.getInstance(), pgp, dbConnection.pgPool)
+            break;
+        case "if-not-exists": {
+            const dbExists = await dbAdminApi.databaseExists()
+            if (dbExists) {
+                requestLogger.info(`Database ${ServerConfig.getInstance().pgDb()} already exists, keep existing database, (config option 'if-not-exists').`)
+            } else {
+                requestLogger.info(`Creating new database ${ServerConfig.getInstance().pgDb()} because it does not exist yet, (config option 'if-not-exists').`)
+                await dbAdminApi.createDatabase()
+            }
+            break;
+        }
+        default: {
+            break;
+        }
+    }
 
-// Initialize database
-if (ServerConfig.getInstance().createDatabase()) {
-    await dbAdminApi.createDatabase()
+    // Initialize repositories
+    const existingRepositories = (await dbAdminApi.listRepositories()).queryResult
+        .map(s => s.schema_name)
+        .filter(s => s.startsWith(SCHEMA_PREFIX))
+        .map(s => s.substring(SCHEMA_PREFIX.length))
+    requestLogger.info("Existing repositories " + existingRepositories)
+    for (const repository of ServerConfig.getInstance().createRepositories()) {
+        const repoCreation = repository.create
+        switch (repoCreation) {
+            case "always":
+                requestLogger.info(`Creating new repository ${repository.name} (config option 'always')`)
+                if (existingRepositories.includes(repository.name)) {
+                    // need to remove the repository first
+                    const deletedn = await  dbAdminApi.deleteRepository({clientId: "setup", repository: SCHEMA_PREFIX + repository.name})
+                    requestLogger.info(`Delete repository ${repository.name} result is ` + JSON.stringify(deletedn))
+                    const newEexistingRepositories = (await dbAdminApi.listRepositories()).queryResult
+                        .map(s => s.schema_name)
+                        .filter(s => s.startsWith(SCHEMA_PREFIX))
+                        .map(s => s.substring(SCHEMA_PREFIX.length))
+                    requestLogger.info("Repositories now are: " + newEexistingRepositories)
+                }
+                await createRepository(repository)
+                break;
+            case "never":
+                requestLogger.info(`Not creating repository ${repository.name} (config option 'never')`)
+                break;
+            case "if-not-exists": {
+                if (existingRepositories.includes(repository.name)) {
+                    requestLogger.info(`Repository ${repository} already exists, keep existing repository, (config option 'if-not-exists').`)
+                } else {
+                    requestLogger.info(`Creating new repository ${repository} because it does not exist yet, (config option 'if-not-exists').`)
+                    await createRepository(repository)
+                }
+                break;
+            }
+            default: {
```

Comment:
I would consider removing the default or alternatively keeping it and throwing an error

Reply:
Removed empty default